src/posix/platform/mainloop.hpp

```diff
@@ -65,6 +65,12 @@ class Source
      */
     virtual void Process(const otSysMainloopContext &aContext) = 0;
 
+    /**
+     * This method marks desturctor virtual.
+     *
+     */
+    virtual ~Source() = default;
```

Comment:
This is only necessary if the class is going to be deleted. see: https://patchwork.ozlabs.org/project/gcc/patch/BANLkTinxuAywtkOsNzSb+Ge6hmPWSm_D4Q@mail.gmail.com/

Reply:
Exactly.
My idea is it's a point of view on class design, it needs to be easier to be used right. If its destructor is virtual, the user of this polynomial class can worry less if it's used correctly.